File: RP_performance/v1.java


// UPI Payment implementation
class UPIPaymentStrategy {
    public void pay() {
        System.out.println("Payment Done Using UPI");
    }
}

// Netbanking Payment implementation
class NetbankingPaymentStrategy {
    public void pay() {
        System.out.println("Payment Done Using Netbanking");
    }
}

// Credit Card Payment implementation
class CreditCardPaymentStrategy {
    public void pay() {
        System.out.println("Payment Done Using Credit Card");
    }
}

// A function to make payment using the given strategy
public class PaymentProcessor {

    public static void makePayment(??? strategy) {
        strategy.pay();
    }

    public static void main(String[] args) {
        PaymentStrategy upi = new UPIPaymentStrategy();
        PaymentStrategy netbanking = new NetbankingPaymentStrategy();
        PaymentStrategy creditCard = new CreditCardPaymentStrategy();

        makePayment(upi);
        makePayment(netbanking);
        makePayment(creditCard);
    }
}

File: RP_performance/v1.cpp
// Online C++ compiler to run C++ program online
#include <iostream>
class UPIPaymentStratergy{
    
    public :
       void pay(){std::cout<<"Payment Done Using UPI"<<std::endl; }
      
};
class NetbankingPaymentStratergy{
    public :
       void pay(){std::cout<<"Payment Done Using Netbanking"<<std::endl; }
    
};
class CreditCardPayamentStratergy{
    public :
       void pay(){std::cout<<"Payment Done Using Credit Card"<<std::endl; }
    
};


void makePayment(??? *paymentStratergy){
    paymentStratergy->pay();
}

int main() {
   UPIPaymentStratergy upi;
   makePayment(&upi);
   
   NetbankingPaymentStratergy nb;
   makePayment(&nb);
   
   CreditCardPayamentStratergy ccp;
   makePayment(&ccp);
   
   

    return 0;
}
